### ISSUE TITLE
Define which errors should raise exceptions

### ISSUE THREAD

**Issue by cknaut** [lukingroup/pylabnet#22]
One of the main features of pylabnet is that errors in sub-routines will not crash the entire experiment by raising exceptions, but rather just output an error message into the `LogHandler`. However, some errors will make the execution of scripts impossible, such as for example errors occurring during setup of `HardwareDrivers`. 

I will for now raise explicit exceptions for all setup-related code but am opening this issue as an reminder to discuss this topic at some point and to define a guideline.

**Comment by mkbhaskar** [comment]
typically, we distinguish between "errors" and "warnings". I wonder if such a hierarchy is possible? No idea how to implement it (I guess we would need to manually designate in our own error handling whether something is an "error" or a "warning"), but this would be great.

Both would be logged, but only "errors" actually raise an exception

**Comment by cknaut** [comment]
We do have the option to designate log-message levels (debug, info, warning, error).  We could tell the logger to raise an exception if the log-level is 'error', but this would raise the exception in the thread of the logger, not the script that actually produces the error. 

What we could do is after sending an error message to the logger manually raise an exception:

 
```python
# Assume this in some HardwareClient class
if some_failing_function():
    error_message = 'This failed'

    # Log error in the LogHandler
    self.log.error(error_message)

    # Raise exception in current thread
    raise Exception(error_message )
```

**Comment by cknaut** [comment]
Or maybe I could write an extension to the `LogHandler` that does this automatically.

**Comment by mkbhaskar** [comment]
Yes, this would be great!

**Comment by cknaut** [comment]
Exception raising of `LogHandler` implemented in 07c1acbd84c30544ef93df55ca09d625bc51e24a.
